react-app/src/app/components/RestrictRoute/index.js

```diff
@@ -3,25 +3,37 @@ import { Route, Redirect } from 'react-router-dom';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 
-// TODO: if the current date is greater than the date tokenExpireDateTime then invalidate token and redirect to login
-// TODO: delete eslint-disable-next-line
+import { getUser } from '../../../redux/Auth/actions';
+
 class RestrictRoute extends Component {
   render() {
-    if (!this.props.logged) {
-      return <Redirect to="/login" />;
+    this.props.getUser(this.props.token);
```

Comment:
Why do you need to get the user in each screen render. This is going to be executed many times